modules/service/src/main/scala/lucuma/odb/service/ProgramService.scala

```diff
@@ -313,72 +286,10 @@ object ProgramService {
               UpdateProgramsError.InvalidSemester(SET.semester.toOption).failure
           }
 
-        // Update proposals. This can fail in a few ways.
-        val updateProposals: F[Result[List[Program.Id]]] =
-          SET.proposal.fold(Result(Nil).pure[F]) {
-            proposalService.updateProposals(_)
-          }
-
-        // A stable identifier (ie. a `val`) is needed for the enums.
-        val enumsVal = enums
-
-        def tagToProposalStatus(tag: Tag): Result[enumsVal.ProposalStatus] =
-          Enumerated[enumsVal.ProposalStatus]
-            .fromTag(tag.value)
-            .fold(UpdateProgramsError.InvalidProposalStatus(user, tag).failure)(Result.apply)
-
-        def userCanChangeProposalStatus(ps: enumsVal.ProposalStatus): Boolean =
-          user.role.access =!= Access.Guest && (ps <= enumsVal.ProposalStatus.Submitted || user.role.access >= Access.Ngo)
-
-        def validateStatusUpdate(
-          pid:             Program.Id,
-          oldStatus:       enumsVal.ProposalStatus,
-          newStatusUpdate: Option[enumsVal.ProposalStatus],
-          hasProposal:     Boolean
-        ): Result[Unit] =
-          newStatusUpdate
-            .filter(_ =!= oldStatus) // if they match we're not really trying to update
-            .fold(Result.unit) { newStatus =>
-              (
-                UpdateProgramsError.NoProposalForStatusChange(pid)
-                  .failure.unlessA(hasProposal),
-
-                UpdateProgramsError.NotAuthorizedNewProposalStatus(user, Tag(newStatus.tag))
-                  .failure.unlessA(userCanChangeProposalStatus(newStatus)),
-
-                UpdateProgramsError.NotAuthorizedOldProposalStatus(pid, user, Tag(oldStatus.tag))
-                  .failure.unlessA(userCanChangeProposalStatus(oldStatus))
-
-              ).tupled.void  // do we want all of the errors or would it be annoying?
-            }
-
-        def validateUpdate(newStatusUpdate: Option[enumsVal.ProposalStatus]): F[Result[Unit]] =
-          session.prepareR(Statements.getTempTableData).use(
-            _.stream(Void, chunkSize = 1024)
-             .fold(Result.unit) { case (acc, (pid, oldSemester, psTag, hasProposal)) =>
-               val check: Result[Unit] =
-                 for {
-                   oldStatus <- tagToProposalStatus(psTag)
-                   _         <- validateStatusUpdate(pid, oldStatus, newStatusUpdate, hasProposal)
-                   finalStatus   = newStatusUpdate.getOrElse(oldStatus)
-                   finalSemester = SET.semester.fold(none, oldSemester, _.some)
-                   _         <- UpdateProgramsError.NoSemesterForSubmittedProposal(pid)
```

Comment:
I think (hope) we would get unhandled constraint violations if we allow people to unset the semester.  Is this handled somewhere else so that it is caught before it gets that far?

Reply:
It is now not possible to unset the semester.